apps/emqx/src/emqx_cm.erl

```diff
@@ -192,14 +190,6 @@ do_unregister_channel({_ClientId, ChanPid} = Chan) ->
     ok = emqx_hooks:run('channel.unregistered', [ChanPid]),
     true.
 
--spec connection_closed(emqx_types:clientid()) -> true.
-connection_closed(ClientId) ->
-    connection_closed(ClientId, self()).
-
--spec connection_closed(emqx_types:clientid(), chan_pid()) -> true.
-connection_closed(ClientId, ChanPid) ->
-    ets:delete_object(?CHAN_CONN_TAB, {ClientId, ChanPid}).
```

Comment:
- `?CHAN_CONN_TAB` table store `{{ClientId, ChanPid, Mod}`, so ets:delete_object do anything here.
- we have already deleted {clientid, ChanPid} in emqx_cm process after ChanPid shutdown here:
https://github.com/emqx/emqx/blob/master/apps/emqx/src/emqx_cm.erl#L686-L692
So we don't need this connection_closed/1 function now.

Reply:
I remember at least the function was left unchanged previously by @savonarola 
but I cannot recall the discussion about why delete_object is still here.

Reply:
I added a clarification in 4.4
https://github.com/emqx/emqx/blob/main-v4.4/src/emqx_cm.erl#L173

>     %% We can't clean CHAN_CONN_TAB because records for dead connections
>     %% are required for `get_chann_conn_mod/1` function, and `get_chann_conn_mod/1`
>     %% is used for takeover.

I left the fun empty to avoid attempts to "fix" the things while we have the current mechanism of keeping offline sessions.